cdap-security/src/main/java/io/cdap/cdap/security/store/FileSecureStoreService.java

```diff
@@ -413,11 +439,41 @@ private SecureStoreData deserialize(byte[] data) throws IOException {
     byte[] secret = new byte[dis.readInt()];
     dis.readFully(secret);
 
+    overwriteSensitiveByteArray(decodedData);
+
     return new SecureStoreData(meta, secret);
   }
 
+  /**
+   * Zeros the given byte array to prevent sensitive data from being persisted in memory.
+   *
+   * @param arr The byte array containing sensitive data
+   */
+  private static void overwriteSensitiveByteArray(byte[] arr) {
+    for (int i = 0; i < arr.length; i++) {
+      arr[i] = 0;
+    }
+  }
+
   private static String getKeyName(final String namespace, final String name) {
-    return namespace + NAME_SEPARATOR + name;
+    // Namespaces can be case-sensitive, while key names cannot. As we cannot assume the backing implementation of
+    // KeyStore supports case-sensitive aliases, the namespace must be hex-encoded during alias generation.
+    return encodeToHex(namespace.getBytes(StandardCharsets.UTF_8)) + NAME_SEPARATOR + name;
+  }
+
+  /**
+   * Encodes a given string into a UTF-8 hex string.
+   * @param bytes The input bytes to encode
+   * @return Hex encoding of the input string
+   */
+  @VisibleForTesting
+  static String encodeToHex(byte[] bytes) {
```

Comment:
There is a `Bytes.toHexString` method already

Reply:
Done